packages/react-native/ReactCommon/react/renderer/components/scrollview/ScrollViewShadowNode.cpp

```diff
@@ -66,7 +66,18 @@ void ScrollViewShadowNode::layout(LayoutContext layoutContext) {
 Point ScrollViewShadowNode::getContentOriginOffset() const {
   auto stateData = getStateData();
   auto contentOffset = stateData.contentOffset;
-  return {-contentOffset.x, -contentOffset.y + stateData.scrollAwayPaddingTop};
+
+  auto props = getConcreteProps();
+
+  float productX = 1.0f;
+  float productY = 1.0f;
+
+  for (const auto& operation : props.transform.operations) {
+    productX *= operation.x;
+    productY *= operation.y;
+  }
```

Comment:
This code seems only ever valid for scale operations. I think we want to multiple the transforms themselves. We need to be multiplying the transform matrices.

Reply:
I've updated it to use the transform matrix. Thanks!